### PR TITLE
Math Magicians-Events

### PR DESCRIPTION
**Pull request by EzekielUtshudi**
### In this PR I implemented the following : 
- [ ] Created the logic/ directory in my project
- [ ] Created calculate.js and operate.js,  ( added [big.js](https://www.npmjs.com/package/big-js) as dependency).
- [ ] Imported the files in your Calculator component.
- [ ] Implemented the event handlers I  need to use the math logic from calculate.js and operate.js in my React component.
- [ ] I useD state and props, pass props to child components, used lifecycle methods, and lift the state up.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import './Calculator.css';
+import PropTypes from 'prop-types';
+
```
Comment (TedLivist) [inline src/components/Calculator.js]: - [x] You have not implemented the events. You need to import the provided `calculate.js` file and use it to implement your calculation events.

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      return one.div(two).toString();
+    } catch (err) {
+      return "Can't divide by 0.";
+    }
+  }
+  if (operation === '%') {
+    return one.mod(two).toString();
+  }
+  throw Error(`Unknown operation '${operation}'`);
+}
```
Comment (TedLivist) [inline src/logic/operate.js]: - [x] Your `operate.js` file is not the same as the provided file. As I can see there is no exception handler for the modulus operation for `0`. Kindly ensure your file is the same as this: [`operate.js file`](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js)

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @EzekielUtshudi,

Good job so far!
✔️ Good documentation
✔️ Provided logic files are imported
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [x] Kindly fix your README project title to be descriptive.

  ezekiel-proj

_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Calculator.js (hunk 1 above)
Comment (BenMukebo) [inline src/components/Calculator.js]: - [x] As required by the previous reviewer, The Calculator component should import the `calculate.js` file from the logic directory and use it to implement your calculation events. kindly do this

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+export default function operate(numberOne, numberTwo, operation) {
+  const one = Big(numberOne);
+  const two = Big(numberTwo);
+  if (operation === '+') {
+    return one.plus(two).toString();
+  }
+  if (operation === '−') {
+    return one.minus(two).toString();
+  }
```
Comment (BenMukebo) [inline src/logic/operate.js]: - [x] As requested by the previous review,  according to the project requirement,  Your `operate.js` file should be the same as the provided file.  Kindly Make sure to copy the entire data from this: [`operate.js file`](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js)

Check 

   |        Yours          |            Original.        |
---------------------:|:--------------------
bad1 | cool1

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @EzekielUtshudi  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 

- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️


 ## Required Changes ♻️
_Check the comments under the review._


You have submitted a project, that Linters check are not running in the GitHub/action
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/operate.js (hunk 3 above)
Comment (Whoistolu) [reply src/logic/operate.js]: Well resolved 👍🏽

File: src/components/Calculator.js (hunk 1 above)
Comment (Whoistolu) [reply src/components/Calculator.js]: Now resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @EzekielUtshudi 

Great job implementing the recommended changes 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**